internal/campaigns/service.go

```diff
@@ -312,7 +312,31 @@ query NamespaceQuery($name: String!) {
 }
 `
 
+const usernameQuery = `
+query {
+    currentUser {
+        id
+    }
+}
+`
+
 func (svc *Service) ResolveNamespace(ctx context.Context, namespace string) (string, error) {
+	if namespace == "" {
+		// if no namespace is provided, default to logged in user as namespace
+		var resp struct {
+			Data struct {
+				CurrentUser struct {
+					ID string `json:"id"`
+				} `json:"currentUser"`
+			} `json:"data"`
+		}
+		if ok, err := svc.client.NewRequest(usernameQuery, nil).DoRaw(ctx, &resp); err != nil || !ok {
+			return "", errors.WithMessage(err, "failed to resolve namespace: no user logged in")
+		}
+
+		return resp.Data.CurrentUser.ID, nil
```

Comment:
Do I need a nil check for `CurrentUser` or `CurrentUser.ID`?

Reply:
Hmm, I think currentUser is nilable. That would mean "not logged in, I guess"

Reply:
Actually, they can't be nil because they aren't pointers, so nil-checks won't even compile. (I'm assuming this is fine as is?)

Reply:
Ah, then I'd either make CurrentUser a pointer, or check for the empty string, but I think it is a valid case that the API returns an empty current user, so I think we should cover that here.
```suggestion
                 if resp.Data.CurrentUser.ID == "" {
                     return nil, errors.New("cannot resolve current user")
                 }
		return resp.Data.CurrentUser.ID, nil
```